rasa/core/policies/memoization.py

```diff
@@ -411,9 +411,9 @@ def _recall_using_truncation(
         logger.debug("Launch DeLorean...")
 
         # Truncate the tracker based on `max_history`
-        truncated_tracker = _trim_tracker_by_max_history(
-            tracker, self.config[POLICY_MAX_HISTORY]
-        )
+        truncated_tracker: Optional[
+            DialogueStateTracker
+        ] = _trim_tracker_by_max_history(tracker, self.config[POLICY_MAX_HISTORY])
```

Comment:
```suggestion
        truncated_tracker: DialogueStateTracker = _trim_tracker_by_max_history(tracker, self.config[POLICY_MAX_HISTORY])
```
should be enough

Reply:
it needs to be marked as `Optional` because the return value of `_strip_leading_events_until_action_executed()` is `Optional[DialogueStateTracker]`

Reply:
(otherwise the type annotation is not needed at all)